src/test/java/seedu/todo/commons/util/DateUtilTest.java

```diff
@@ -119,8 +156,110 @@ public void formatTimeTests() {
+        assertTrue(DateUtil.checkIfDateExist(currentDate.plusMonths(1)));
+        assertTrue(DateUtil.checkIfDateExist(currentDate.plusMonths(12)));
+        //only day is diff
+        assertTrue(DateUtil.checkIfDateExist(currentDate.plusDays(1)));
+        assertTrue(DateUtil.checkIfDateExist(currentDate.plusDays(365))); 
+               
+        //day and month are diff
+        currentDate = currentDate.plusDays(1);
+        currentDate = currentDate.plusMonths(1);
+        assertTrue(DateUtil.checkIfDateExist(currentDate));
+        
+        //day and year diff
+        currentDate = currentDate.minusMonths(1);
+        currentDate = currentDate.plusYears(1);
+        assertTrue(DateUtil.checkIfDateExist(currentDate));
+        
+        //month and year diff
+        currentDate = currentDate.minusDays(1);
+        currentDate = currentDate.plusMonths(1);
+        assertTrue(DateUtil.checkIfDateExist(currentDate));
+    }
+    
+    //@@author A0139922Y
+    @Test
+    public void testCheckIfTimeExist_false() {
+        LocalDateTime currentTime = LocalDateTime.now();
+        assertNotNull(DateUtil.checkIfTimeExist(currentTime));
+        assertFalse(DateUtil.checkIfTimeExist(currentTime));
+    }
+    
+    //@@author A0139922Y
+    @Test
+    public void testIfTimeExist_true() {
+        LocalDateTime currentTime = LocalDateTime.now().toLocalDate().atTime(10,0);
+        assertNotNull(DateUtil.checkIfTimeExist(currentTime));
+        assertTrue(DateUtil.checkIfTimeExist(currentTime.toLocalDate().atTime(currentTime.getHour() - currentTime.getHour(), 
+                currentTime.getMinute())));
+        assertTrue(DateUtil.checkIfTimeExist(currentTime.toLocalDate().atTime(currentTime.getHour(), 
+                currentTime.getMinute() - currentTime.getMinute())));
+        assertTrue(DateUtil.checkIfTimeExist(currentTime.toLocalDate().atTime(currentTime.getHour() - currentTime.getHour(), 
+                currentTime.getMinute() - currentTime.getMinute())));
+    }
+    
+    //@@author A0139922Y
+    @Test
+    public void testParseTimeStamp() {
+        LocalDateTime currentDay = LocalDateTime.now();
+        LocalDateTime currentDayAt2Pm = currentDay.toLocalDate().atTime(2, 0);
+        LocalDateTime nextDay = currentDay.plusDays(1);
+        LocalDateTime nextDayAt2Pm = nextDay.toLocalDate().atTime(2, 0);
+        LocalDateTime nextDayAt2359 = nextDay.toLocalDate().atTime(23, 59);
+        assertNotEquals(DateUtil.parseTimeStamp(currentDay, null, true), currentDay);
+        assertEquals(DateUtil.parseTimeStamp(currentDay, null, true), DateUtil.floorDate(currentDay));
+        //will set the time to 00:00
+        assertNotEquals(DateUtil.parseTimeStamp(currentDay, nextDay, true), currentDay);
+        assertEquals(DateUtil.parseTimeStamp(currentDay, nextDay, true), DateUtil.floorDate(currentDay));
+        // will set the time to 23:59
+        assertNotEquals(DateUtil.parseTimeStamp(nextDay, currentDay, false), nextDay);
```

Comment:
it is definitely not advisable to have more then one assert, tests should also be focused, unless really needed 1 or 2 asserts per test is plausible, to have more then 10 or even 5 is already really off. Please refactor.